### ISSUE TITLE
Create objects at the forest level vs. tree level

### ISSUE THREAD

**Issue by motiwari** [ThrunGroup/scikit-learn#2]
In our last meeting, one of the TODOs was: file an issue for whether the objects (was it the histograms?) can be allocated at the forest-level or tree-level.

@RyanK0902 do you remember what object this was?

**Comment by RyanK0902** [comment]
Yes it was the `_BinMapper` class defined [here](https://github.com/scikit-learn/scikit-learn/blob/8175cd60e3df606905aedf2dd3ba0832e50cd963/sklearn/ensemble/_hist_gradient_boosting/binning.py#L70).
